console/src/view/resources.rs

```diff
@@ -0,0 +1,189 @@
+use crate::{
+    state::{
+        resources::{Resource, SortBy},
+        State,
+    },
+    view::{
+        self, bold,
+        table::{self, TableList, TableListState},
+        DUR_LEN, DUR_PRECISION,
+    },
+};
+
+use tui::{
+    layout,
+    style::{self, Color, Style},
+    text::Spans,
+    widgets::{Cell, Paragraph, Row, Table},
+};
+
+#[derive(Debug, Default)]
+pub(crate) struct ResourcesTable {}
+
+impl TableList for ResourcesTable {
+    type Row = Resource;
+    type Sort = SortBy;
+
+    const HEADER: &'static [&'static str] = &[
+        "ID",
+        "Kind",
+        "Total",
+        "Target",
+        "Type",
+        "Location",
+        "Attributes",
+    ];
+
+    fn render<B: tui::backend::Backend>(
+        table_list_state: &mut TableListState<Self>,
+        styles: &view::Styles,
+        frame: &mut tui::terminal::Frame<B>,
+        area: layout::Rect,
+        state: &mut State,
+    ) {
+        let now = if let Some(now) = state.last_updated_at() {
+            now
+        } else {
+            // If we have never gotten an update yet, skip...
+            return;
+        };
+
+        table_list_state
+            .sorted_items
+            .extend(state.resources_state_mut().take_new_resources());
+        table_list_state
+            .sort_by
+            .sort(now, &mut table_list_state.sorted_items);
+
+        let dur_cell = |dur: std::time::Duration| -> Cell<'static> {
+            Cell::from(styles.time_units(format!(
+                "{:>width$.prec$?}",
+                dur,
+                width = DUR_LEN,
+                prec = DUR_PRECISION,
+            )))
+        };
```

Comment:
nit/TIOLI: this is copied from the tasks list view, right? it _could_ be worth making it a function in `view`...but, then we would have to pass in `Styles` every time we call it instead of capturing it as a closure. so, maybe this is actually nicer...

Reply:
...actually, i see that we only have one duration for resources currently. maybe this doesn't really need to be a function here, we could just do it once to format the total time. i think it makes sense for it to be a function in the tasks list since it's repeated a bunch.